slick/src/main/scala/slick/jdbc/SQLiteProfile.scala

```diff
@@ -95,12 +95,24 @@ trait SQLiteProfile extends JdbcProfile {
 
   class ModelBuilder(mTables: Seq[MTable], ignoreInvalidDefaults: Boolean)(implicit ec: ExecutionContext) extends JdbcModelBuilder(mTables, ignoreInvalidDefaults) {
     override def createColumnBuilder(tableBuilder: TableBuilder, meta: MColumn): ColumnBuilder = new ColumnBuilder(tableBuilder, meta) {
-      /** Regex matcher to extract name and length out of a db type name with length ascription */
-      final val TypePattern = "^([A-Z\\s]+)(\\(([0-9]+)\\))?$".r
-      private val (_dbType,_size) = meta.typeName match {
-        case TypePattern(d,_,s) => (d, Option(s).map(_.toInt))
-        case "" => ("TEXT", None)
+      /** Regex matcher 
+        * to extract name and length 
+        * out of a db type name with length ascription 
+        */
+      def extractTypeProps( 
+        s: String = "NUMERIC(18,2)"
+      , pattern: scala.util.matching.Regex = "^([A-Z\\s]+)(\\(([0-9]+),?([0-9]+)?\\))?$".r
+      ): ( String, Option[ Int ] ) = s match {
+        case pattern( 
+          name, numericProps, precision, scale 
+        ) => ( name, Option( precision ).map( _.toInt ) )
+        case "" => ( "TEXT", None )
       }
+      private 
+      val ( _dbType, _size ) = extractTypeProps( 
+        s = meta.typeName
+      , pattern = "^([A-Z\\s]+)(\\(\\s?([0-9]+)\\s?,?\\s?([0-9]+)?\\s?\\))?$".r 
```

Comment:
@GlulkAlex, I don't think it is needed to have the pattern as a parameter to the `extractTypeProps`. 

We could keep it as:

```scala
final val TypePattern = "^([A-Z\\s]+)(\\(\\s?([0-9]+)\\s?,?\\s?([0-9]+)?\\s?\\))?$".r

def extractTypeProps(typeName: String = "NUMERIC(18,2)"): (String, Option[Int]) =
  typeName match {
    case TypePattern(name, _, precision, _) => (name, Option(precision).map(_.toInt) )
    case "" => ("TEXT", None)
  }
```

The return type `ColumnBuilder` and that method only exists on this specific anonymous instance and therefore is practically the equivalent of a `private` method. 

Since we have currently a merge conflict, and I offered myself to fix it, I will do that small change. Just wanted to let you know. Please, let me know if there were any particular reason for having it as a parameter. I may be missing something.

Reply:
@renatocaval As I can recall 
there is nothing special about having RegEx pattern as a parameter . 
It just looked more explicit and functional to me .

Reply:
ok, fair enough. 

That's all very local, internal scope, so I prefer to simplify it as it will not be used in any other place.